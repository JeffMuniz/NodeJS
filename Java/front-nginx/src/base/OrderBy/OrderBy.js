import React from "react";
import { arrayOf, func, string, shape } from "prop-types";
import { Option, Container, Span } from "./OrderBy.styles";

const OrderByComponent = ({ options, onChange, selectedOption, title }) => (
  <Container>
    <Span>{title}: </Span>
    {options.map(option => (
      <Option
        id={option.label}
        key={option.label}
        selected={selectedOption === option.value}
        onClick={onChange}
      >
        <span>{option.label}</span>
      </Option>
    ))}
  </Container>
);

OrderByComponent.propTypes = {
  options: arrayOf(
    shape({
      label: string,
      value: string,
    }),
  ).isRequired,
  onChange: func.isRequired,
  selectedOption: string.isRequired,
  title: string,
};

OrderByComponent.defaultProps = {
  title: "",
};

export default OrderByComponent;
