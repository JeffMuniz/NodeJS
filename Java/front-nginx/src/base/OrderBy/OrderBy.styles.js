import styled from "styled-components";

import { red, gainsboro } from "@colors";

export const Option = styled.div`
  border: 1px solid ${props => (props.selected ? red : gainsboro)};
  color: ${props => (props.selected ? red : gainsboro)}
  border-radius: 38px;
  padding: 7px;
  min-width: 115px;
  text-align: center;
  display: inline-block;
  cursor: pointer;
  margin-right: 10px;
  font-size: 14px;
  font-weight: 300;
  user-select: none;
`;

export const Container = styled.div`
  display: flex;
  flex: 1;
  align-items: center;
`;

export const Span = styled.span`
  margin-right: 25px;
`;
