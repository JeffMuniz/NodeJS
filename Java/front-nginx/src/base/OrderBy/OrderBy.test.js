import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import OrderBy from "./OrderBy";

describe("OrderBy", () => {
  it("Should render it", () => {
    // given
    const renderer = new ShallowRenderer();
    const props = {
      options: [
        {
          label: "",
          shape: "",
          value: "",
        },
      ],
      onChange: jest.fn(),
      selectedOption: "",
    };

    // when
    renderer.render(<OrderBy {...props} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
