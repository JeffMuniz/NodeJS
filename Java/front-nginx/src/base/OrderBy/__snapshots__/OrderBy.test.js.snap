// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderBy Should render it 1`] = `
<styled.div>
  <styled.span>
    
    : 
  </styled.span>
  <styled.div
    id=""
    onClick={[MockFunction]}
    selected={true}
  >
    <span>
      
    </span>
  </styled.div>
</styled.div>
`;
